source/features/restore-file.tsx

```diff
@@ -53,42 +60,60 @@ async function restoreFile(progress: (message: string) => void, menuItem: Elemen
 		throw new Error('Restore failed: File too big');
 	}
 
-	let {pathname} = menuItem.previousElementSibling as HTMLAnchorElement;
-	// Check if file was deleted by PR
-	if (menuItem.closest('[data-file-deleted="true"]')) {
-		progress('Undeleting…');
-		const [nameWithOwner, headBranch] = select('.head-ref')!.title.split(':');
-		pathname = `/${nameWithOwner}/new/${headBranch}?filename=${filePath}`;
-	} else {
-		progress('Committing…');
-	}
+	const [nameWithOwner, prBranch] = select('.head-ref')!.title.split(':');
+	progress(menuItem.closest('[data-file-deleted="true"]') ? 'Undeleting…' : 'Committing…');
 
 	const content = file.text;
-	// This is either an `edit` or `create` form
-	const form = (await fetchDom(pathname, 'form.js-blob-form'))!;
-	form.elements.value.value = content; // Restore content (`value` is the name of the file content field)
-	form.elements.message.value = (form.elements.message as HTMLInputElement).placeholder
-		.replace(/^Create|^Update/, 'Restore');
-
-	const response = await pushForm(form);
-	if (!response.ok) {
-		throw new Error(response.statusText);
-	}
+	await api.v4(`mutation {
+		createCommitOnBranch(input: {
+			branch: {
+				repositoryNameWithOwner: "${nameWithOwner}",
+				branchName: "${prBranch}"
+			},
+			expectedHeadOid: "${await getHeadReference()}",
+			fileChanges: {
+				additions: [
+					{
+						path: "${filePath}",
+						contents: "${btoa(unescape(encodeURIComponent(content)))}"
+					}
+				]
+			},
+			message: {
+				headline: "Restore ${filePath}"
+			}
+		}) {
+			commit {
+				oid
+			}
+		}
+	}`);
 }
 
+const filesRestored = new WeakSet<HTMLButtonElement>();
 async function handleRestoreFileClick(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
 	const menuItem = event.delegateTarget;
 
-	await showToast(async progress => {
+	// Only allow one click
+	if (filesRestored.has(menuItem)) {
+		return;
+	}
+
+	filesRestored.add(menuItem);
```

Comment:
We can drop this too. It's part of the old logic that kept the menu open to display the status. Thanks to `showToast`, double-clicks are impossible now